src/omotes_sdk/omotes_interface.py

```diff
@@ -200,26 +233,31 @@ def connect_to_submitted_job(
             auto_disconnect_handler,
         )
 
+        # TODO: raise an error if job queues are not found when the client reconnects.
```

Comment:
I was initially thinking to use [get_queue](https://aio-pika.readthedocs.io/en/latest/apidoc.html#aio_pika.Channel.get_queue) method to first check if the queue already exists and raises an error if not, but that would need to turn `connect_to_submitted_job()` into an async function? and I don't feel like doing so as it seems to introduce a change for the SDK users.

Alternatively, might to be set the [passive](https://aio-pika.readthedocs.io/en/latest/apidoc.html#aio_pika.Channel.declare_queue) flag when trying to _resume_ the queue connection, but that is going to be a few levels deep in broker interface `_declare_queue()`, so also a bit hesitant.

Curious if there is any suggestion.

Reply:
Why would you expect that we need to turn `connect_to_submitted_job()` into an async function?

To show how an exception travels in the current code path:
The function, used by `connect_to_submitted_job()`, to connect to a queue is `declare_queue_and_add_subscription`. This function calls an async function but also adds the `.result()` call to ensure that any result (or exception) is retrieved and the synchronuous call will only terminate if the asynchronuous call has finished (due to the `.result()`). I would expect if async `_declare_queue_and_add_subscription` throws an error/exception, it would bubble up all the way through `declare_queue_and_add_subscription()` and `connect_to_submitted_job()`.

All-in-all, if you would use `aio_pika.get_queue` in a new function e.g. async def `_get_queue_and_add_subscription` and throw an exception when needed I would expect it to work. Then you can add a sync def `get_queue_and_add_subscription` similarly to how `declare_queue_and_add_subscription` is setup.

Reply:
Thanks for helping out! I add and use `queue_exists()` to check if the queue exists. This check will be carried out when SDK tries to reconnect via `connect_to_submitted_job()` and `reconnect` is True. If the queue does not exist, an error is raised and returned, and the reconnection is stopped.